src/validation/rules/optional/ProhibitDeprecatedFieldsRule.js

```diff
@@ -0,0 +1,52 @@
+// @flow strict
+
+import { GraphQLError } from '../../../error/GraphQLError';
+
+import { type EnumValueNode, type FieldNode } from '../../../language/ast';
+import { type ASTVisitor } from '../../../language/visitor';
+
+import { getNamedType } from '../../../type/definition';
+
+import { type ValidationContext } from '../../ValidationContext';
+
+/**
+ * Prohibit deprecated fields
+ *
+ * A GraphQL document is only valid if all selected fields and all used enum
+ * values have not been deprecated.
+ *
+ * Note: This rule is optional and is not part of the Validation section of the
+ * GraphQL Specification.
+ */
+export function ProhibitDeprecatedFieldsRule(
+  context: ValidationContext,
+): ASTVisitor {
+  return {
+    Field(node: FieldNode) {
+      const fieldDef = context.getFieldDef();
+      const parentType = context.getParentType();
+      if (parentType && fieldDef && fieldDef.deprecationReason != null) {
+        context.reportError(
+          new GraphQLError(
+            `The field ${parentType.name}.${fieldDef.name} is deprecated. ` +
+              fieldDef.deprecationReason,
+            [node],
```

Comment:
You don't need to wrap node in the array, you actually did it bellow with enum error.